src/world/sprite.c

```diff
@@ -25,13 +25,19 @@
 #include "fountain.h"
 #include "sprite.h"
 
-rct_sprite* g_sprite_list = RCT2_ADDRESS(RCT2_ADDRESS_SPRITE_LIST, rct_sprite);
-
 uint16 *gSpriteListHead = RCT2_ADDRESS(RCT2_ADDRESS_SPRITE_LISTS_HEAD, uint16);
 uint16 *gSpriteListCount = RCT2_ADDRESS(RCT2_ADDRESS_SPRITE_LISTS_COUNT, uint16);
 
 uint16 *gSpriteSpatialIndex = (uint16*)0xF1EF60;
 
+rct_sprite* sprite_list = RCT2_ADDRESS(RCT2_ADDRESS_SPRITE_LIST, rct_sprite);
+
+rct_sprite *get_sprite(size_t sprite_idx)
+{
+	assert(sprite_idx < MAX_SPRITES);
```

Comment:
I could be easily wrong here, but doesn't this assert get triggered always as the maximum number of sprites is always higher then the number of sprites in the save-file? Or am I reading the code wrong?